attiny/attiny.ino

```diff
@@ -123,36 +129,40 @@ void requestEvent() {
 }
 
 void receiveEvent(uint8_t howMany) {
-  howMany--;
-  byte h;
-  byte l;
-  switch(TinyWireS.receive()) {
+  if (howMany < 1 || howMany > sizeof(message)) {
+    // Message too short or too long.
+    return;
+  }
+  memset(message, 0x00, sizeof(message));
+  int i = 0;
+  while(howMany--) {
+    message[i++] = TinyWireS.receive();
+  }
+  newMessage = true;
+}
+
+void readi2cMessage() {
+  newMessage = false;
+  switch(message[0]) {
     case 0x11:
+      blinks = 5;
       minuteCountdown = MINUTE_COUNTDOWN;
-      h = TinyWireS.receive();
-      howMany--;
-      l = TinyWireS.receive();
-      howMany--;
-      cli();
-      piSleepTime = (h << 8) + l;
-      sei();
+      piSleepTime = (message[1] << 8) + message[2];
```

Comment:
Whenever you're reading or writing a volatile variable that's bigger than a byte that might get updated in an interrupt handler (ISR), you need to disable interrupts while you're interacting with that variable. The processor needs to use multiple instructions when dealing with multi-byte variables and an interrupt can happen between them. If interrupts are disabled you can sometimes end up with incomplete reads.

As far I can see, this goes for `message` and `piSleepTime`.